server.js

```diff
@@ -29,23 +29,28 @@ app.use(function(req, res, next) {
 app.post('/items', function (req, res) {
 	if (!req.body) return res.sendStatus(400);
 
-	var item = req.body;
-
-	Item.where('name', item.name).fetch().then(function(results){
-		if (results === null) {
-			Item.forge({name: item.name, quantity: item.quantity}).save().then(function(r) {
-				var itemId = r.get('id');
-				PurchasedItem.forge({item_id: itemId, quantity: item.quantity, unit: 0, cost: 0}).save().then(function(r) {
-					if (r) console.log('POST add item');
+	var itemList = req.body;
+	var count = 0;
+	itemList.forEach(function(item) {
+		Item.where('name', item.name).fetch().then(function(results){
+			if (results === null) {
+				Item.forge({name: item.name, quantity: item.quantity}).save().then(function(r) {
+					var itemId = r.get('id');
+					PurchasedItem.forge({item_id: itemId, quantity: item.quantity, unit: 0, cost: 0}).save().then(function(r) {
+						if (r) console.log('post add item');
+					});
 				});
-			});
-		} else {
-			Item.forge().where({name: item.name}).save({quantity: parseInt(results.get('quantity'))+parseInt(item.quantity)}, { method: 'update' }).then(function(r){
-				var itemId = results.get('id');
-				PurchasedItem.forge({item_id: itemId, quantity: item.quantity, unit: 0, cost: 0}).save().then(function(r) {
-					if (r) console.log('POST add item');
+			} else {
+				Item.forge().where({name: item.name}).save({quantity: parseInt(results.get('quantity'))+parseInt(item.quantity)}, { method: 'update' }).then(function(r){
+					var itemId = results.get('id');
+					PurchasedItem.forge({item_id: itemId, quantity: item.quantity, unit: 0, cost: 0}).save().then(function(r) {
+						if (r) console.log('POST add item');
+					});
 				});
-			});
+			}
+		});
+		if (++count === itemList.length) {
+			return res.send('Success');
```

Comment:
the problem with this is that we don't know if all of the callbacks have executed at this point, it will probably return before everything is saved